PyRDF/Node.py

```diff
@@ -79,4 +61,11 @@ def _graph_prune(self):
             
             return False
 
+        elif self.operation and not self.next_nodes and (self.operation.op_type == Operation.Types.TRANSFORMATION):
```

Comment:
Why don't we need to check the number of references for a transformation node?

Reply:
This is a great question. @etejedor , It wouldn't work if the user tries to re-use the node which was pruned. It'd throw this error : 
```python
---> 33         return getattr(self.action_node.value, self._cur_attr)(*args, **kwargs)
AttributeError: 'NoneType' object has no attribute 'GetValue'
```
This is also another reason why we should not prune the leaf nodes which are transformations. In the current implementation, the only way for the user to fix this is by redeclaring all the necessary nodes (which were pruned). Shall I change it back to the older implementation then ?
@etejedor , @bluehood ?

Reply:
That would be a problem then, I suggest that we do not prune transformations to which the user still has a reference, unless @bluehood has some objection?
It is worth noting that, with this code, transformations with no reference from the user are pruned just like any other node, unless they have children (or grandchildren, etc.) that are still referenced by the user. Is this correct @shravan97 ?

Reply:
Yes @etejedor . It'd prune the transformations without any references anyway  :+1: